klee/lib/Core/Executor.cpp

```diff
@@ -450,6 +450,36 @@ Executor::StatePair Executor::fork(ExecutionState &current, const ref<Expr> &con
     return StatePair(trueState, falseState);
 }
 
+Executor::StatePair Executor::fork(ExecutionState &current) {
+    if (current.forkDisabled) {
+        return StatePair(&current, nullptr);
+    }
+
+    ExecutionState *clonedState;
+    notifyBranch(current);
+    clonedState = current.clone();
+    addedStates.insert(clonedState);
+
```

Comment:
At first, I thought this is enough. However, it simply gives me a segfault.

So I inspected how `klee::ExecutionState::clone()` is implemented. At line 72, `state->concolics` is now managing a brand new instance of `klee::Assignment`.

https://github.com/S2E/s2e/blob/bf2b660f06955a51b67df25546784d51f2c9770c/klee/lib/Core/ExecutionState.cpp#L69-L74

Here we can see that a new shared_ptr and a new instance of `klee::Assignment` is created.

https://github.com/S2E/s2e/blob/bf2b660f06955a51b67df25546784d51f2c9770c/klee/include/klee/util/Assignment.h#L69-L71

I ended up recomputing concrete values (just like the regular fork() does) and everything works fine now.

Reply:
What segfault does it give?